### PR TITLE
feat(ci): add automatic docs generation

### PR DESCRIPTION
**Pull request by ariscript**
## Confirmation
- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines](https://github.com/nullishamy/JVM/blob/main/CONTRIBUTING.MD)

### Changes

- [ ] Internal
- [ ] Feature
- [ ] Documentation
- [x] Other: CI actions workflow

Closes Issue: #22 

## Description

Add an actions workflow to generate documentation (using `cargo doc`) and host it on GitHub pages. Runs on the same triggers as the release workflow, and uses the code from the tagged commit. There's a preview available on my repository's page: https://dheerajpv.github.io/JVM/

### CODE REVIEW

**Comment by nullishamy** [comment]
Should we consider renaming the crate / repo so that the URL fits better..? I had been considering this anyways, and now seems opportune.

**Comment by nullishamy** [comment]
Also please target `dev`. You can simply change the branch in the GH UI because my `main` is up to date.

**Comment by nullishamy** [comment]
I will update the GH pages config to use the main branch's docs/ folder as we will only want to host stable docs. I will also update the original issue to reflect this change in plan.

**Comment by ariscript** [comment]
> Should we consider renaming the crate / repo so that the URL fits better..? I had been considering this anyways, and now seems opportune.

Yes, though we would need to rename both because index.html purely just redirects to `/jvm` (which is the crate name)

**Comment by nullishamy** [comment]
I'm struggling with good names so if you have any feel free to comment.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                  override: true
+
+            - name: Delete existing docs directory
+              run: rm -rf ./docs/
+
+            - name: Generate docs
+              run: cargo doc --release --no-deps
+
+            - name: Create index.html for github pages
+              run: echo "<meta http-equiv=\"refresh\" content=\"0; url=jvm\">" > target/doc/index.html
```
Comment (nullishamy) [inline .github/workflows/docs.yml]:
```suggestion
              run: echo "<meta http-equiv=\"refresh\" content=\"0; url=Kate\">" > target/doc/index.html
```

**Comment by nullishamy** [comment]
lol macos autocorrect made the K capital, that's wrong. could you accept the new suggestion?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                  override: true
+
+            - name: Delete existing docs directory
+              run: rm -rf ./docs/
+
+            - name: Generate docs
+              run: cargo doc --release --no-deps
+
+            - name: Create index.html for github pages
+              run: echo "<meta http-equiv=\"refresh\" content=\"0; url=Kate\">" > target/doc/index.html
```
Comment (nullishamy) [inline .github/workflows/docs.yml]:
```suggestion
              run: echo "<meta http-equiv=\"refresh\" content=\"0; url=kate\">" > target/doc/index.html
```

**Comment by nullishamy** [comment]
Alrighty, lets see if this works on my GH pages site.

**Pull request merged**